src/components/PopupWithForm.js

```diff
@@ -6,12 +6,23 @@ export default class PopupWithForm extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._popupFormInputs =
       this._popupElement.querySelectorAll(".modal__input");
+    this._popupButton = this._popupElement.querySelector(".modal__button");
     this._handleFormSubmit = handleFormSubmit;
     this._handleSubmitListener = (evt) => {
       evt.preventDefault();
+      this._popupButton.textContent = "Saving...";
       const inputValues = this._getInputValues();
-      this._handleFormSubmit(inputValues);
-      this.close();
+
+      this._handleFormSubmit(inputValues)
+        .then(() => {
+          this.close();
+          setTimeout(() => {
+            this._popupButton.textContent = "Save";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.